app/models/miq_request.rb

```diff
@@ -191,12 +191,15 @@ def must_have_user
   end
 
   def call_automate_event_queue(event_name)
+    user = User.current_user
     MiqQueue.put(
       :class_name  => self.class.name,
       :instance_id => id,
       :method_name => "call_automate_event",
       :args        => [event_name],
       :zone        => options.fetch(:miq_zone, my_zone),
+      :user_id     => user.try(:id),
```

Comment:
@lfu You need to validate that the User.current_user is set, you just can't pass a null into Automate.